### PR TITLE
internal/auth: migrate httpheader middlware logging

### PR DESCRIPTION
**Pull request by davejrt**
Morning effort as part of https://github.com/sourcegraph/sourcegraph/issues/40896

## Test plan

Green tests on PR

### CODE REVIEW

File: enterprise/cmd/frontend/internal/auth/httpheader/middleware.go [hunk 1]
```diff
@@ -43,14 +43,15 @@ func Middleware(db database.DB) *auth.Middleware {
...
+			logger := log.Scoped("middleware", "check http header")
 			authProvider, ok := providers.GetProviderByConfigID(providers.ConfigID{Type: providerType}).(*provider)
 			if !ok {
 				next.ServeHTTP(w, r)
 				return
 			}
 
 			if authProvider.c.UsernameHeader == "" {
-				log15.Error("No HTTP header set for username (set the http-header auth provider's usernameHeader property).")
+				logger.Error("No HTTP header set for username (set the http-header auth provider's usernameHeader property).")
```
Comment (jhchabran) [inline enterprise/cmd/frontend/internal/auth/httpheader/middleware.go:54]: I think this and the others should be `Warn` instead, we aren't going to take action on such a sentry report.

**Comment by burmudar** [comment]
oh I migrated this too 😆

**Comment by davejrt** [comment]
> oh I migrated this too 😆

https://media.giphy.com/media/UyWXNumwNvXmE/giphy.gif

**Comment by davejrt** [comment]
Closing in favor of https://github.com/sourcegraph/sourcegraph/pull/41441

**Pull request closed without merging**